src/Container/Class.hs

```diff
@@ -164,245 +176,279 @@ instance ToList IS.IntSet where
+    {-# INLINE minimum #-}
+    all = BS.all
+    {-# INLINE all #-}
+    any = BS.any
+    {-# INLINE any #-}
+    find = BS.find
+    {-# INLINE find #-}
+    head = fmap fst . BS.uncons
+    {-# INLINE head #-}
 
 instance Container BSL.ByteString where
-  foldr = BSL.foldr
-  {-# INLINE foldr #-}
-  foldl = BSL.foldl
-  {-# INLINE foldl #-}
-  foldl' = BSL.foldl'
-  {-# INLINE foldl' #-}
-  foldr1 = BSL.foldr1
-  {-# INLINE foldr1 #-}
-  foldl1 = BSL.foldl1
-  {-# INLINE foldl1 #-}
-  length = fromIntegral . BSL.length
-  {-# INLINE length #-}
-  elem = BSL.elem
-  {-# INLINE elem #-}
-  notElem = BSL.notElem
-  {-# INLINE notElem #-}
-  maximum = BSL.maximum
-  {-# INLINE maximum #-}
-  minimum = BSL.minimum
-  {-# INLINE minimum #-}
-  all = BSL.all
-  {-# INLINE all #-}
-  any = BSL.any
-  {-# INLINE any #-}
-  find = BSL.find
-  {-# INLINE find #-}
-  head = fmap fst . BSL.uncons
-  {-# INLINE head #-}
+    foldr = BSL.foldr
+    {-# INLINE foldr #-}
+    foldl = BSL.foldl
+    {-# INLINE foldl #-}
+    foldl' = BSL.foldl'
+    {-# INLINE foldl' #-}
+    foldr1 = BSL.foldr1
+    {-# INLINE foldr1 #-}
+    foldl1 = BSL.foldl1
+    {-# INLINE foldl1 #-}
+    length = fromIntegral . BSL.length
+    {-# INLINE length #-}
+    elem = BSL.elem
+    {-# INLINE elem #-}
+    notElem = BSL.notElem
+    {-# INLINE notElem #-}
+    maximum = BSL.maximum
+    {-# INLINE maximum #-}
+    minimum = BSL.minimum
+    {-# INLINE minimum #-}
+    all = BSL.all
+    {-# INLINE all #-}
+    any = BSL.any
+    {-# INLINE any #-}
+    find = BSL.find
+    {-# INLINE find #-}
+    head = fmap fst . BSL.uncons
+    {-# INLINE head #-}
 
 instance Container IS.IntSet where
-  foldr = IS.foldr
-  {-# INLINE foldr #-}
-  foldl = IS.foldl
-  {-# INLINE foldl #-}
-  foldl' = IS.foldl'
-  {-# INLINE foldl' #-}
-  length = IS.size
-  {-# INLINE length #-}
-  elem = IS.member
-  {-# INLINE elem #-}
-  maximum = IS.findMax
-  {-# INLINE maximum #-}
-  minimum = IS.findMin
-  {-# INLINE minimum #-}
-  head = fmap fst . IS.minView
-  {-# INLINE head #-}
+    foldr = IS.foldr
+    {-# INLINE foldr #-}
+    foldl = IS.foldl
+    {-# INLINE foldl #-}
+    foldl' = IS.foldl'
+    {-# INLINE foldl' #-}
+    length = IS.size
+    {-# INLINE length #-}
+    elem = IS.member
+    {-# INLINE elem #-}
+    maximum = IS.findMax
+    {-# INLINE maximum #-}
+    minimum = IS.findMin
+    {-# INLINE minimum #-}
+    head = fmap fst . IS.minView
+    {-# INLINE head #-}
+
+----------------------------------------------------------------------------
+-- Wrapped List
+----------------------------------------------------------------------------
+-- | This can be useful if you want to use 'Container' methods for your data type
+-- but you don't want to implement all methods of this type class for that.
+newtype WrappedList f a = WrappedList (f a)
+
+type instance Element (WrappedList f a) = a
+
+instance ToList (f a) => ToList (WrappedList f a) where
+    toList (WrappedList l) = toList l
```

Comment:
Could you please also implement `null` method for `WrappedList` and add `{-# INLINE #-}` pragma for both methods?